Version 1.2.4

ricsal1 opened the pull request:
-   Support for IA Plugin with programmable rules (plugin in development)
-   Protection against game exceptions
-   Fixed scoreboard update error when starting new game
-   Game limited to 16 players due to scoreboard limit (and you can't see all the players)
-   Removed speed potion effects, using direct speed
-   Fixed resource pack load every time you start game by supporting status Downloaded
-   Kill during lobby will not count anymore as kills or win
-   Start time counter directly linked to Shoptime/2, this way can be adjusted
-   Fixed Bukkit and Spigot detection
-   More code cleanup, variable protection
-   Fixed random spawn
-   More general error handling
-   Less spam when missing players to start

ricsal1 committed:
Version 1.0.5
-   Support for IA Plugin with programable rules (plugin in development)
-   Protection against game exceptions
-   Fixed scoreboard update error when starting new game
-   Game limited to 16 players due to scorboard limit (and you can't see all the players)
-   Removed speed potion efects, using direct speed
-   Fixed resource pack load every time you start game by supporting status Downloaded
-   Kill during lobby will not count anymore as kills or win
-   Start time counter directly linked to Shoptime/2, this way can be adjusted
-   Fixed Bukkit and Spigot detection
-   More code cleanup, variable protection
-   Fixed random spawn
-   More general error handling
-   Less spam when missing players to start

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -704,9 +750,13 @@ public void setupPlayers() {
-          //  player.getInventory().setItem(8, getShopItem()); //check PlayerUpdater
+            //  player.getInventory().setItem(8, getShopItem()); //check PlayerUpdater
+
+            try { //npc give exception
+                player.setHealth(40);
+            } catch (Exception e) {
+            }
```
Instead of making a try/catch here, it could be implemented differently, so changing npcs health would work.
Couple of lines above there is `Player player = csplayer.getPlayer();` - it could be changed to sth like `LivingEntity entity = csplayer.getLivingEntity();`

then entity.setHealth(40) should work.

Btw. player.setGameMode, setFlying and setAllowFlight do not throw an error when called od npc?

Another solution would be just to check the type of entity and act accordingly - if it is not a player, do not apply such methods.

ricsal1 replied:
Hi, player.setGameMode, setFlying and setAllowFlight  are ok for NPCs

Will try your LivingEntity  sujestion

ricsal1 replied:
Solved, it wasn't related to npc was about setting GENERIC_MAX_HEALTH first

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -738,12 +788,14 @@ public void setupPlayers() {
         if (!terrorists.isEmpty()) {
             CSPlayer playerWithBomb = (CSPlayer) terrorists.toArray()[new Random().nextInt(terrorists.toArray().length)];
             playerWithBomb.getPlayer().getInventory().setItem(4, CSUtil.getBombItem());
+            Utils.debug("------> Bomb with player "+ playerWithBomb.getPlayer().getName() + "    " +new Random().nextInt(terrorists.toArray().length) + "     "+ new Random().nextInt(terrorists.toArray().length));
```
Creating new Random() each time is wrong for performance reasons. Instead, save it to a variable and reuse it in this method.

ricsal1 replied:
the idea is to force it to "choose" a new player every round, store it will always assign the bomb to the same player

barpec12 replied:
new Random() creates new generator of random numbers. One generator can create many random numbers. There should be no difference to the outcome.
When you call
`Random generator = new Random();`
`generator.nextInt(5); `
`generator.nextInt(5);`

Each next call to generator.nextInt(); will generate independent random number, these numbers will be chosen randomly each time, it most often will not assign the same player. will generate an independent random number; these numbers will be chosen randomly each time.

For additional source see https://app.deepsource.com/directory/analyzers/java/issues/JAVA-W1015

ricsal1 replied:
Roger, thanks for the explanation m8

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -779,8 +831,18 @@ public CSPlayer getCSPlayer(Player player, boolean create, String colour) {
+
+            //online filters NPCs
+            if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {
+                ResourseHash.remove(player.getName() + "RES");
+                ResourseHash.put(player.getName() + "RES", "QUALITY");
```
There is no need to first remove and then use put. Put already will overwrite item with the same key.

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -779,8 +831,18 @@ public CSPlayer getCSPlayer(Player player, boolean create, String colour) {
+            CSPlayer csp = new CSPlayer(this, player, colour);
+
+            //online filters NPCs
+            if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {
```
It is better to use .equals() for String comparisons. see https://www.geeksforgeeks.org/difference-between-and-equals-method-in-java/

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -822,12 +884,17 @@ public GameState getGameState() {
         Location loc = this.getTerroristSpawn(false);
 
         for (Entity ent : loc.getWorld().getEntities()) {
             if (!(ent instanceof Player) && (ent.getType().equals(Material.TNT) || ent.getName().equals("TNT"))) {
```
Often Java programmers use a little trick here. Instead of using myVariable.equals("TNT"), it may be used "TNT".equals(myVariable). Advantage of this solution is when myVariable is null. With the order you have chosen (which is obviously natural to choose when writing code), you may encounter NullpointerExceptions. If the order would be reversed, it would just compare TNT.equals(null), so it would return false without an exception.

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -1006,7 +1112,8 @@ public void returnPlayertoGame(CSPlayer csplay) {
 
-        if (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT") {
+        //isOnline filters NPCs
+        if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {
```
Same as before, use .equals().
Btw, for readability you could extract `ResourseHash.get(player.getName() + "RES"` to a variable - in this method it is used 4 times.

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java:
```diff
@@ -1006,7 +1112,8 @@ public void returnPlayertoGame(CSPlayer csplay) {
-        if (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT") {
+        //isOnline filters NPCs
+        if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {
             ResourseHash.remove(player.getName() + "RES");
             ResourseHash.put(player.getName() + "RES", "QUALITY");
```
Same as before, remove not needed when using put

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/commands/CounterStrikeCommand.java:
```diff
@@ -126,24 +139,68 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
+            double z = loc.getZ();
+            float yaw = loc.getYaw();
+            float pitch = loc.getPitch();
+            String location = world + "," + x + "," + y + "," + z + "," + yaw + "," + pitch;
```
You could use here `String.join(",", world, x, y, z, yaw, pitch)`

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -122,16 +109,18 @@ public CSPlayer(CounterStrike plugin, Player player, String colour) {
         if (player.getInventory().getItemInMainHand().getType().equals(Material.IRON_AXE)) {
-            CounterStrike.i.myBukkit.runTask(player,null,null, () -> player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 99999, (int) (player.getWalkSpeed() * 0.7))));
+          //  CounterStrike.i.myBukkit.runTask(player, null, null, () -> player.addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 99999, (int) (player.getWalkSpeed() * 0.7))));
+            player.setWalkSpeed((float)0.233);
```
```suggestion
            player.setWalkSpeed(0.233f);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -122,16 +109,18 @@ public CSPlayer(CounterStrike plugin, Player player, String colour) {
         } else {
-            CounterStrike.i.myBukkit.runTask(player,null,null, () -> player.removePotionEffect(PotionEffectType.SPEED));
+          //  CounterStrike.i.myBukkit.runTask(player, null, null, () -> player.removePotionEffect(PotionEffectType.SPEED));
+            player.setWalkSpeed((float)0.2);
```
```suggestion
            player.setWalkSpeed(0.2f);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/csplayer/CSPlayer.java:
```diff
@@ -153,8 +142,13 @@ public void clear() {
+
+        if (player != null) {
+            Utils.debug("Clearing CSPlayer data for " + player.getName() + "   " + test1 + "    " + test2 + "    " + test3);
+            player.setWalkSpeed((float)0.2);
```
```suggestion
            player.setWalkSpeed(0.2f);

```

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/listeners/PlayerJoinListener.java:
```diff
@@ -179,20 +162,19 @@ public void playerChangedWorldEvent(PlayerChangedWorldEvent event) {
-                //clears QUALITY resoursepack and loads default
-                if (plugin.ResourseHash.get(player.getName() + "RES") == null || plugin.ResourseHash.get(player.getName() + "RES") == "QUALITY") {
+                //clears QUALITY resoursepack and loads default (inOnline filters NPCs)
+                if (player.isOnline() &&( plugin.ResourseHash.get(player.getName() + "RES") == null || plugin.ResourseHash.get(player.getName() + "RES") == "QUALITY")) {
```
Use equals instead of == and the same as before with remove/put below

barpec12 commented on CounterStrike/src/dev/danablend/counterstrike/utils/MyBukkit.java:
```diff
@@ -224,4 +250,160 @@ public void playerSetResourcePack(Player player, String resourse, String hash) {
+                int pointer = pagina.indexOf("project-file-name-container-" + projectName);
+                pagina = pagina.substring(pointer); //smaller data
+
+                String tmp = pagina.substring(pagina.indexOf("https://cdn.modrinth.com/"));
+                String version = tmp.substring(tmp.indexOf("data-name=\"") + 11).split("\"")[0];
+                String url = tmp.split("\"")[0];
+                String features = tmp.substring(tmp.indexOf("features=\"") + 10).split("\"")[0];
```
I'm not sure what is happening here, but I think it should be done using regex instead.

barpec12 reviewed (approved):
I made some quick review, mostly the fact of using == instead of .equals may lead to bugs, I think.

ricsal1 committed:
Version 1.0.7
-   Support for IA Plugin with programable rules (plugin in development)
-   Better support for 1.20.1
-   Fixed bug when messaging players outside csmc game
-   Added 2 more commands:

      setMap for multi to choose the game map you are configuring
      setbombsite for future AI Plugin purposes

-   Protection against game exceptions/ better error handling
-   Fixed scoreboard update error when starting new game
-   Game limited to 16 players due to scorboard limit (and you can't see all the players)
-   Removed speed potion efects, using direct speed
-   Fixed resource pack load every time you start game by supporting status Downloaded
-   Kill during lobby will not count anymore as kills or win
-   Start time counter directly linked to Shoptime/2, this way can be adjusted
-   Fixed Bukkit and Spigot detection
-   More code cleanup, variable protection
-   Fixed random spawn
-   Less spam when missing players to start
-   Added version checker (every 6h will look for news)

ricsal1 committed:
Version 1.0.7
-   Support for IA Plugin with programable rules (plugin in development)
-   Better support for 1.20.1 and 1.20.2
-   Fixed bug when messaging players outside csmc game
-   Added 2 more commands:

      setMap for multi to choose the game map you are configuring
      setbombsite for future AI Plugin purposes

-   Protection against game exceptions/ better error handling
-   Fixed scoreboard update error when starting new game
-   Game limited to 16 players due to scorboard limit (and you can't see all the players)
-   Removed speed potion efects, using direct speed
-   Fixed resource pack load every time you start game by supporting status Downloaded
-   Kill during lobby will not count anymore as kills or win
-   Start time counter directly linked to Shoptime/2, this way can be adjusted
-   Fixed Bukkit and Spigot detection
-   More code cleanup, variable protection
-   Fixed random spawn
-   Less spam when missing players to start
-   Added version checker (every 6h will look for news)

ricsal1 committed:
Version 1.2.0
 Added modeValorant
  different presentation names
  9000k money limmit
  20 max health
  Inventory names

 Added showGameStatusTitle (to disable big letters popups in begin rounds and end)
 Added standardTeamColours config to enable you to select directly if you want to be a terror (red or gold) or ct (blue, aqua)
 Centralized configs for better future code management
 New stop command to end game
 Added map vote option

ricsal1 committed:
Version 1.2.4
  Added damaged based on configured values, if not with zoom
  Added different player speed for heavy weapons
  Fixed error with previous version
  Dead players now follow the killer
